PGM/src/main/java/tc/oc/pgm/shop/ShopParser.java

```diff
@@ -0,0 +1,158 @@
+        this.filterParser = filterParser;
+        this.translator = translator;
+        this.kitParser = kitParser;
+    }
+
+    @Override
+    public void parse() throws InvalidXMLException {
+        Element shopsRoot = document.getRootElement().getChild("purchase");
+        if (shopsRoot == null)
+            return;
+
+        if (shopsRoot.getChild("item-sets") == null)
+            throw new InvalidXMLException("No item sets are defined.", shopsRoot);
+        if (shopsRoot.getChild("currencies") == null)
+            throw new InvalidXMLException("No currencies are defined.", shopsRoot);
+        if (shopsRoot.getChild("shops") == null)
+            throw new InvalidXMLException("No shops are defined.", shopsRoot);
+
+        for (Element currency : shopsRoot.getChild("currencies").getChildren()) {
+            context.features().define(currency, parseCurrency(currency));
+        }
+
+        for (Element itemSet : shopsRoot.getChild("item-sets").getChildren()) {
+            context.features().define(itemSet, parseItemSet(itemSet));
+        }
+
+        boolean crossTeam = XMLUtils.parseBoolean(shopsRoot.getAttribute("cross-team"), false);
+        boolean persistent = XMLUtils.parseBoolean(shopsRoot.getAttribute("persistent"), false);
+        this.tracker = new SimplePurchaseTracker(crossTeam, persistent);
+
+        for (Element shop : shopsRoot.getChild("shops").getChildren()) {
+            context.features().define(shop, parseShop(shop));
+        }
+    }
+
+    private PurchasableSet parseItemSet(Element el) throws InvalidXMLException {
+        Set<Purchasable> purchasables = Sets.newHashSet();
+        for (Element element : el.getChildren()) {
+            purchasables.add(new PurchasableKit(new ItemCreator(XMLUtils.parseMaterial(Node.fromAttr(element, "icon"))),
+                    integerParser.parse(Node.fromRequiredAttr(element, "slot")),
+                    doubleParser.parse(Node.fromRequiredAttr(element, "price")),
+                    context.features().reference(Node.fromRequiredAttr(element, "currency"), Currency.class),
+                    typeParser.parse(Node.fromRequiredAttr(element, "scope")),
+                    XMLUtils.parseBoolean(Node.fromRequiredAttr(element, "incremental"), false),
+                    filterParser.property(element, element.getAttribute("purchase-filter").getValue()).optional(new StaticFilter(Filter.QueryResponse.ALLOW)),
+                    kitParser.property(element, "kit").required()));
+        }
+        return new PurchasableSet.Impl(purchasables);
+    }
+
+    private Currency parseCurrency(Element el) throws InvalidXMLException {
+        Node node = new Node(el);
+        switch (el.getName()) {
+            case "material":
+                Material material = XMLUtils.parseMaterial(node);
+                Node nameSingle = Node.fromAttr(el, "name-singular");
+                Node namePlural = Node.fromAttr(el, "name-plural");
```

Comment:
That doesn't work for most material names, though. 
e.g.:
Dirts
Stones
Bedrocks
Grasses
Glasses, etc, etc